src/karts/controller/skidding_ai.cpp

```diff
@@ -1965,23 +1972,69 @@ void SkiddingAI::computeNearestKarts()
             -m_world->getOverallDistance(m_kart_behind->getWorldKartId());
     }
 
-    // Compute distance to nearest player kart
-    float max_overall_distance = 0.0f;
+    // Compute distance to target player kart
+
+    float target_overall_distance = 0.0f;
+    float own_overall_distance = m_world->getOverallDistance(m_kart->getWorldKartId());
+    m_num_players_ahead = 0;
+
     unsigned int n = ProfileWorld::isProfileMode()
                    ? 0 : race_manager->getNumPlayers();
+
+    std::vector<float> overall_distance;
+    // Get the players distances
     for(unsigned int i=0; i<n; i++)
     {
-        unsigned int kart_id =
-            m_world->getPlayerKart(i)->getWorldKartId();
-        if(m_world->getOverallDistance(kart_id)>max_overall_distance)
-            max_overall_distance = m_world->getOverallDistance(kart_id);
+        unsigned int kart_id = m_world->getPlayerKart(i)->getWorldKartId();
+        overall_distance.push_back(m_world->getOverallDistance(kart_id));
+    }
+
+    // Sort the list (the list is very short, so a basic bubble sort suffice)
+    for(unsigned int i=0; i<n-1; i++)
```

Comment:
Even so, please don't reinvent the wheel. Just use one of the standard sorting methods